googlesource/googlesource.go

```diff
@@ -90,26 +95,35 @@ func (s *Source) ReadGoogleRow(rowInput chan readRowInput, responseCh chan sdk.R
 				break
 			}
 			if err != nil {
-				sdk.Logger(s.Ctx).Error().Str("err", err.Error()).Msg("error while iterating")
+				sdk.Logger(s.ctx).Error().Str("err", err.Error()).Msg("error while iterating")
 				return err
 			}
 
+			// haris: does BQ have its own way of tracking rows, i.e. its own offsets?
+			// Neha: Could not find any. Tables metadata does not provide any such info.
+			// Users generally have some keys to do so. And we are working on meta-data of
+			// table and not actual data.
```

Comment:
@hariso I could find `iterator.PageInfo().Token` which can be used to get the next page data. But it works for pagination and not on each record. So say we pull **500** records in a page from Bigquery but were able to process only **200** due to some technical issue. The next time we start pulling we will get a token for the next page which is **501** and not **201**.

Reply:
In that case, you can use the previous token, no? i.e. you don't save a page token until you're done with it.

Reply:
then we might sync some rows twice. If I take the above example - we would start pulling from 1 again. And again a similar issue of data inconsistency.

Reply:
It looks like this discussion isn't relevant anymore, since we're doing the other approach (incrementing ID/timestamp field).